[Python][Build] Fix building openvino wheel on Windows

OrestChura opened the pull request:
### Details:

- adds INTEL_CPU condition to inference_engine_snippets target
- adds snippets dependency to snippetsNgraphFunctions target
- manages snippetsNgraphFunctions linkage to tests

When I was trying to build static OpenVINO on Windows, it failed with the following error:

```
 Building Python wheel openvino-2023.0.0-9483-cp39-cp39-win_amd64.whl
  C:\Program Files\Python39\lib\site-packages\setuptools\_distutils\dist.py:264: UserWarning: Unknown distribution option: 'build'
    warnings.warn(msg)
  running bdist_wheel
  Set default value for CMAKE_BUILD_TYPE = Release.
  running build
  running build_clib
  cmake --install C:/git/openvino/build-x86_64 --prefix libs.core --config Release --strip --component core
  Create build directory: build\temp.win32-3.9\Release
  Installing ie_libs
  -- Installing: C:/git/openvino/build-x86_64/src/bindings/python/wheel/libs.core/runtime/lib/intel64/Release/pugixml.lib
  -- Installing: C:/git/openvino/build-x86_64/src/bindings/python/wheel/libs.core/runtime/lib/intel64/Release/itt.lib
  -- Installing: C:/git/openvino/build-x86_64/src/bindings/python/wheel/libs.core/runtime/lib/intel64/Release/util.lib
  CMake Error at C:/git/openvino/build-x86_64/src/common/snippets/cmake_install.cmake:39 (file):
    file INSTALL cannot find
    "C:/git/openvino/bin/intel64/Release/inference_engine_snippets.lib": File
    exists.
  Call Stack (most recent call first):
    C:/git/openvino/build-x86_64/src/common/cmake_install.cmake:52 (include)
    C:/git/openvino/build-x86_64/src/cmake_install.cmake:53 (include)
    C:/git/openvino/build-x86_64/cmake_install.cmake:47 (include)
```

Commands used:

```bat
cmake ^
    -D BUILD_SHARED_LIBS=OFF ^
    -D ENABLE_LTO=ON ^
    -D ENABLE_TESTS=OFF ^
    -D ENABLE_FUNCTIONAL_TESTS=OFF ^
    -D ENABLE_INTEL_CPU=OFF ^
    -D ENABLE_INTEL_GPU=OFF ^
    -D ENABLE_ONEDNN_FOR_GPU=OFF ^
    -D ENABLE_INTEL_GNA=OFF ^
    -D ENABLE_AUTO=OFF ^
    -D ENABLE_AUTO_BATCH=OFF ^
    -D ENABLE_HETERO=OFF ^
    -D ENABLE_MULTI=OFF ^
    -D ENABLE_TEMPLATE=OFF ^
    -D ENABLE_IR_V7_READER=OFF ^
    -D ENABLE_OV_ONNX_FRONTEND=OFF ^
    -D ENABLE_OV_PADDLE_FRONTEND=OFF ^
    -D ENABLE_OV_TF_FRONTEND=OFF ^
    -D ENABLE_GAPI_PREPROCESSING=OFF ^
    -D ENABLE_OV_IR_FRONTEND=ON ^
    -D ENABLE_TBBBIND_2_5=OFF ^
    -D CMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded ^
    -D ENABLE_OPENCV=OFF ^
    ..
    cmake --build . --config Release --parallel
```

### Tickets:
 - *ticket-id*

OrestChura committed:
Add snippets dependency

ilya-lavrenov commented on src/bindings/python/wheel/CMakeLists.txt:
```diff
@@ -28,7 +28,7 @@ set(PUGIXML_LIBS_DIR runtime/3rdparty/pugixml/lib)
 #
 
-foreach(_target ie_api constants _pyngraph openvino_c pyopenvino ov_plugins ov_frontends py_ov_frontends)
+foreach(_target ie_api constants _pyngraph openvino_c pyopenvino ov_plugins ov_frontends py_ov_frontends inference_engine_snippets)
```
`inference_engine_snippets` is a dependency of Intel CPU plugin.
If CPU plugin is not built as in your configuration, we should not build snippets as well.

I suppose we need to add `if(ENABLE_INTEL_CPU)` here https://github.com/openvinotoolkit/openvino/blob/master/src/common/CMakeLists.txt#L9

OrestChura replied:
Thank you so much for the review and suggestions!

I didn't quite understand one thing: you advise to add a condition for building snippets, right?  
But, if they are not built, - wouldn't the wheel build crush in that case?  
Or, do you mean this condition will remove an expected dependency from the wheel as well, so there will be no error?

Anyway, I'll experiment with that locally and come back with the results

ilya-lavrenov replied:
> Or, do you mean this condition will remove an expected dependency from the wheel as well, so there will be no error?

Yes

OrestChura replied:
@ilya-lavrenov thank you so much for the advise!
Checked locally, your way works perfectly.

OrestChura committed:
Merge branch 'master' into oc/fix_wheel_win

OrestChura committed:
- removed dependency back
- added an INTEL_CPU condition on snippets configuring -> no dependency when configured w/0 CPU

ilya-lavrenov committed:
Merge branch 'master' into oc/fix_wheel_win

OrestChura commented:
Builds on Linux w/o INTEL_CPU shows these changes are not enough:
```
-- Configuring done
CMake Error at cmake/developer_package/add_ie_target.cmake:104 (target_include_directories):
  Error evaluating generator expression:

    $<TARGET_PROPERTY:inference_engine_snippets,SOURCE_DIR>

  Target "inference_engine_snippets" not found.
Call Stack (most recent call first):
  src/tests/ngraph_helpers/snippets_ngraph_functions/CMakeLists.txt:10 (addIeTarget)
```

Looking into it

OrestChura committed:
Disable snippets_ngraph_functions conditionally if inference_engine_snippets are not configured

OrestChura committed:
Merge branch 'master' into oc/fix_wheel_win

OrestChura commented:
@ilya-lavrenov @ilyachur @IvanNovoselov
It was also necessary to add a condition for snippetsNgraphFunctions target
And after checking trailing dependencies - there should be changes done to shared tests target linkage
